query/src/sql/optimizer/m_expr.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Debug;
+use std::fmt::Formatter;
+
+use common_exception::Result;
+
+use crate::sql::optimizer::memo::Memo;
+use crate::sql::optimizer::pattern_extractor::PatternExtractor;
+use crate::sql::optimizer::rule::RulePtr;
+use crate::sql::optimizer::rule::TransformState;
+use crate::sql::optimizer::s_expr::PlanPtr;
+use crate::sql::optimizer::SExpr;
+use crate::sql::IndexType;
+
+/// `MExpr` is abbreviation of multiple expression, which is the representation of relational
+/// expressions inside `Memo`.
+#[derive(Clone)]
+pub struct MExpr {
```

Comment:
Maybe rename to `GroupExpression` is better?

Reply:
It was named in `GroupExpression` at the beginning. But I thought the word `Expression` is too common in databend, `MExpr` and `SExpr` looked more distinct for me.